staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -36,6 +37,15 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
+// StartUntil starts f in a new goroutine and calls done once f has finished.
+func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
```

Comment:
Thanks for the name suggestion! I don't want to encumber call sites with currying (however it's not my call) - please take a look at the convenience functions I came up with. WDYT?